### ISSUE TITLE
Unclear error message when an argument matcher is instantiated and not used

### ISSUE THREAD

**Issue by wu46** [dart-lang/mockito#219]
Scenario 1:

in my test main.dart:
```dart
import 'package:http/http.dart' as http;
import 'package:mockito/mockito.dart';
import 'package:test/test.dart';

void main() {
  test('unused local variable', (){
    final mockClient = MockHttpClient();
    final matcher = argThat(isA<http.Request>());

    when(mockClient.send(matcher)).thenAnswer(
        (_) async => http.StreamedResponse(Stream.empty(), 400)
    );
  });
}

class MockHttpClient extends Mock implements http.Client {}
```

Running this dummy test passes.

Scenario 2:
Create the `matcher` local variable but forget to use it.
```dart
import 'package:http/http.dart' as http;
import 'package:mockito/mockito.dart';
import 'package:test/test.dart';

void main() {
  test('unused local variable', (){
    final mockClient = MockHttpClient();
    final matcher = argThat(isA<http.Request>());

    when(mockClient.send(argThat(isA<http.Request>()))).thenAnswer(
        (_) async => http.StreamedResponse(Stream.empty(), 400)
    );
  });
}

class MockHttpClient extends Mock implements http.Client {}
```

I get this error:

```
Invalid argument(s): An argument matcher (like `any`) was used as a named argument, but did not use a Mockito "named" API. Each argument matcher that is used as a named argument needs to specify the name of the argument it is being used in. For example: `when(obj.fn(x: anyNamed("x")))`.
package:mockito/src/mock.dart 332:7                                             _InvocationForMatchedArguments._reconstitutePositionalArgs
package:mockito/src/mock.dart 249:31                                            new _InvocationForMatchedArguments
package:mockito/src/mock.dart 217:18                                            _useMatchedInvocationIfSet
package:mockito/src/mock.dart 114:18                                            Mock.noSuchMethod
package:http/src/client.dart 141:28                                             MockHttpClient.send
```

This took me a long time to track down, as I thought I was using arg matchers incorrectly. Maybe change it to a more helpful message like
'arg matcher was created but never matched' or something?

**Comment by natebosch** [comment]
This looks like the same thing as https://github.com/dart-lang/mockito/issues/215

The line `final matcher = argThat(...);` won't work, it needs to be within a `when` or `verify` call.

This will be more clear after https://github.com/dart-lang/mockito/pull/216 gets published.

**Comment by natebosch** [comment]
I've published as `4.1.1`. If you `pub upgrade` you should get a more clear error message here.
